resources/assets/components/CampaignIdSingle/index.js

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { map, isEmpty } from 'lodash';
+
+import './campaignidsingle.scss';
+
+import Action from '../Action';
+import UploaderModal from '../UploaderModal';
+// import CreateActionModal from '../CreateActionModal';
```

Comment:
@DFurnes @katiecrane I'm a little stuck trying to get the action modal to pop up when clicking the "ADD ACTION" button (for reference, [here](https://projects.invisionapp.com/share/DEQ8H2DXN2W#/screens/343746016) is the mock!). 

If I comment out this line, everything loads as it should on the page. But when this line is commented back in, it seems like it loads the `CreateActionModal` component before loading this page. I'm not sure why because the state should be `false` and not render this until it is called when the button is clicked but I'm thinking maybe something got out of sync and it thinks it should load the `CreateActionModal` component first? Is there anything that I'm missing here?

Reply:
hm another thing that is happening is that when this is commented in, `/campaign-ids` doesn't load either with this error: `Uncaught TypeError: Cannot convert undefined or null to object`. When it is commented out, everything is fine. This makes me further think that something with the sequence of components being loaded is off with this `CreateActionModel` but I can't see anything obvious why it is off!

Reply:
@chloealee are you able to see the state and if `displayCreateActionModal` is actually set as `true` or `false` in the case where it's borked and loads that component too soon?

Reply:
@katiecrane I'm not! So when `import CreateActionModal from '../CreateActionModal';` in commented IN, and I `console.log` the `state`, it shows up as `null`. But when it is commented OUT and I `console.log` the `state`, `displayCreateActionModal` is `false`. So thats why I'm thinking for some reason it is skipping right over this page and going straight to the modal first?

Reply:
weirddddd @chloealee can you try console logging something maybe in the constructor of each component and see if that shows anything? Maybe some higher up stuff too to try to see where it's getting stuck. You might need to use the "preserve logs" option if you end up getting redirected or something

Reply:
thanks @katiecrane !! Really appreciate you taking a look at this! It DOES seem to be going straight to the `CreateActionModal` when I `console.log` something in both the constructors :thinking_face: also still wondering why it breaks the `CampaignIdOverview` page when the `import CreateActionModal from '../CreateActionModal';` is commented in so maybe it is something higher that I'm not importing correctly or something?

Reply:
hmmmm I wonder if the constructor gets hit when it is imported? and then something with that errors out so it never makes it farther in this file?? @chloealee wait is the `CampaignIdOverview` page a totally separate page?

Reply:
@katiecrane it is! That is why it is strange why it is even hit!

Reply:
is it ok to keep this in here to continue iterating on this when I have time? I feel like it is so close!

Reply:
@chloealee yes I think that makes sense!!